src/imitation/scripts/common/expert.py

```diff
@@ -0,0 +1,51 @@
+"""Common configuration elements for loading of expert policies."""
+import sacred
+from huggingface_sb3 import EnvironmentName, ModelName, ModelRepoId, load_from_hub
+from stable_baselines3 import PPO
+from stable_baselines3.common import policies
+
+from imitation.policies import base
+from imitation.scripts.common.common import common_ingredient, make_venv
+
+expert_ingredient = sacred.Ingredient("expert", ingredients=[common_ingredient])
+
+
+@expert_ingredient.config
+def config():
+    policy_type = "ppo"  # right now only ppo supported
+    policy_path = None  # path to zip-file containing an expert policy
+    huggingface_repo_id = None
+    hugginface_orga = "HumanCompatibleAI"
+
+    locals()  # quieten flake8
+
+
+@expert_ingredient.capture
+def get_huggingface_repo_id(common, huggingface_repo_id, policy_type, hugginface_orga):
+    if huggingface_repo_id is not None:
+        return huggingface_repo_id
+    else:
+        return ModelRepoId(
+            hugginface_orga,
+            ModelName(policy_type, EnvironmentName(common["env_name"])),
+        )
+
+
+@expert_ingredient.capture
+def get_expert_path(common, policy_path, policy_type):
+    if policy_path is not None:
+        return policy_path
+    else:
+        model_name = ModelName(policy_type, EnvironmentName(common["env_name"]))
+        return load_from_hub(get_huggingface_repo_id(), model_name.filename)
+
+
+@expert_ingredient.capture
+def get_expert_policy(policy_type) -> policies.BasePolicy:
+    env = make_venv()
+    if policy_type == "ppo":
```

Comment:
Looking further I see you've ripped out all the calls to `load_policy`. Unfortunately that leads to a fairly substantial regression in functionality. E.g. we no longer support SAC, although that's easy to fix. More significantly, there's no way for an end user to define new policy types, unlike with the policy registry.

I'm all for simplifying this, `imitation.policies.serialize` was a bit of a mess, but I fear this might be too simple. Ideally the logic for deserialization would also not be tightly coupled with the scripts, so that people can load policies programatically elsewhere.

Reply:
Curious to hear others thoughts e.g. @norabelrose @scottemmons @yawen-d in terms of features we'd like to have for loading policies.

Reply:
I tried to make the expert ingredient use the `serialize` module again but then we do really run into the issues with the degenerate cases (zero/random policies) and it becomes messy again.

In the end we want the functionality to "turn a bunch of strings (model type, path, hf-repo-id, hf-orga) into some policy". Right now this is scattered between the expert ingredient and `serialize`. We need to if we either want to place all of it in one or the other:

## Option 1: Everything goes to `serialize`:

In this case the expert ingredient becomes a really thin wrapper around `serialize`. It is just about plugging the right configuration values into the right loader function.
In addition to the `ppo` and `sac` type that we register in the registry right now, we would also add `ppo-huggingface` and `sac-huggingface` types. The path, that is passed for the factory function then becomes the huggingface repo id or the env id (we have to figure out what is most convenient).
**Pro:** Powerful model loading features, that can be used without sacred while still being extendable by the user.
**Con:** `serialize` becomes more complex than it already is.

## Option 2: Everything goes to the expert ingredient

In this case the `serialize` module would become so thin, that we might even be able to remove it entirely (no registry would be needed any more).
To let the user extend the types of loadable experts, they could add new sub-ingredients to the expert ingredient:

```python
@expert_ingredient.config
def config():
    type: str = None  # The type of expert to load

@expert_ingredient.capture
def load(type):
    for i in expert_ingredient.ingredients:
        if i.path == type:
            return i.load()

@ppo_expert_ingredient.config
def config():
    path: str = None # The path to load the PPO policy from

@ppo_expert_ingredient.config
def load(path):
    return PPO.load(path).policy

expert_ingrdient.ingrdients.append(ppo_expert_ingrdient)
```

**Pros:** 
- Nudge users to use sacred and reuse sacred ingredients. 
- New policy types can introduce their own configuration values without changing the code in `imitation`. 
- No more degenerate cases (e.g. random policy which has to ignore the path).
**Con:** 
- We enforce the usage of sacred
- Adding a new policy type requires writing a new sacred ingredient which is harder than just registering a callback.

Which solution would you prefer @AdamGleave , @norabelrose , @scottemmons , @yawen-d ?

I would be in favor of Option 2, since it seems less hacky. But maybe you have other options in mind?

Reply:
I think there's two key questions here:
  1. Do we want the serialization to work just from scripts, or also from ad-hoc Python code?
  2. Do we want to tie ourselves more deeply to Sacred specifically?

For 1) I lean slightly towards allowing ad-hoc Python code, for research frameworks flexibility is valuable as hard to anticipate use cases. But I don't feel strongly about this, ["There's Only One Way To Do it"](https://wiki.python.org/moin/TOOWTDI) has its benefits too for consistency across projects.

For 2) though I'd really like to avoid having Sacred too deeply integrated. The idea of Sacred was great. But at this point it's largely unmaintained. There are known serious issues with it that would require a significant redesign to fix and seem unlikely to occur. At some point I'd like to survey the other options and potentially migrate to something else, although there's nothing that's obviously better. Still, this pushes me towards trying to keep the Sacred side as thin as possible.

Would definitely appreciate other people's thoughts here!

I do like the design of option 2 allowing you to specify different configs for different policy types. Maybe we can keep that part of the design even with serialize.py; i.e. move from a type signature of `load_policy(type: str, path: str)` to `load_policy(type: str, **kwargs)`?

Reply:
Turns out that my proposed option 2 was not so easy to implement after all. To avoid too much entanglement with `sacred` I implemented your proposal about modifying the interface of `load_policy`. I am happy with the solution, `expert.py` became rather short now. Am happy to have your eyes on this again @AdamGleave